src/pool.jl

```diff
@@ -206,6 +251,9 @@ function levels!(pool::CategoricalPool{S, R}, newlevels::Vector;
 
     empty!(pool.invindex)
     resize!(pool.levels, n)
+    pool.hash = hashlevels(levs)
```

Comment:
I would set it to `0` (assuming that `hash` does not produce `0`) and then add the following function:
```
function gethash(a::CategoricalPool)
    if a.hash == 0
        a.hash = hashlevels(levs)
    end
    return a.hash
end
```

Reply:
of course this change affects all other places.